src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -181,7 +185,61 @@ private CaseData setApplicantDefenceResponseDocFlag(CaseData caseData) {
     }
 
     private CallbackResponse setApplicantRouteFlags(CallbackParams callbackParams) {
-        CaseData updatedCaseData = setApplicantDefenceResponseDocFlag(setApplicant1ProceedFlagToYes(callbackParams.getCaseData()));
+        CaseData caseData = callbackParams.getCaseData();
+        CaseData updatedCaseData = setApplicantDefenceResponseDocFlag(setApplicant1ProceedFlagToYes(caseData));
+
+        /*((caseData.getRespondent1ClaimResponseTypeForSpec() == RespondentResponseTypeSpec.FULL_DEFENCE
+            && caseData.getApplicant1ProceedWithClaim() == YES)
+            || caseData.getApplicant1AcceptAdmitAmountPaidSpec() == NO) {
+            Set<DefendantResponseShowTag> showConditionFlags = caseData.getShowConditionFlags();
+            showConditionFlags.add(DefendantResponseShowTag.VULNERABILITY);
+            updatedCaseData.toBuilder().showConditionFlags(showConditionFlags);
+        }*/
+
+       setMediationConditionFlag(updatedCaseData);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.toMap(objectMapper))
+            .build();
+    }
+
+    private void setMediationConditionFlag(CaseData updatedCaseData) {
+        if(isMediationRequired(updatedCaseData).equals(YES)) {
+            Set<DefendantResponseShowTag> showConditionFlags = updatedCaseData.getShowConditionFlags();
+            showConditionFlags.add(DefendantResponseShowTag.MEDIATION_2);
+            updatedCaseData.toBuilder().showConditionFlags(showConditionFlags);
+        }
+    }
+
+    private YesOrNo isMediationRequired(CaseData caseData) {
```

Comment:
change to boolean as this method will not populate to ccd

Reply:
done